navigation/libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/models/RouteStepProgressTest.java

```diff
@@ -0,0 +1,132 @@
+      = new RouteStepProgress(firstLeg, 0, Mockito.mock(Position.class));
+    assertNotNull("should not be null", routeStepProgress);
+  }
+
+  @Test
+  public void getStepDistance_equalsZeroOnOneCoordSteps() throws Exception {
+    Gson gson = new Gson();
+    String body = readPath(DCMAPBOX_CHIPOLTLE);
+    response = gson.fromJson(body, DirectionsResponse.class);
+    DirectionsRoute route = response.getRoutes().get(0);
+    RouteLeg lastLeg = route.getLegs().get(0);
+    int totalStepCountInLeg = lastLeg.getSteps().size() - 1;
+
+    RouteStepProgress routeStepProgress =
+      new RouteStepProgress(lastLeg, totalStepCountInLeg, Mockito.mock(Position.class));
+    assertNotNull("should not be null", routeStepProgress);
+    assertEquals(1, routeStepProgress.getFractionTraveled(), DELTA);
+    assertEquals(0, routeStepProgress.getDistanceRemaining(), DELTA);
+    assertEquals(0, routeStepProgress.getDistanceTraveled(), DELTA);
+    assertEquals(0, routeStepProgress.getDurationRemaining(), DELTA);
+  }
+
+  @Test
+  public void getDistanceRemaining_equalsStepDistanceAtBeginning() {
+    LineString lineString
+      = LineString.fromPolyline(firstLeg.getSteps().get(5).getGeometry(), Constants.PRECISION_6);
+    double stepDistance = TurfMeasurement.lineDistance(lineString, TurfConstants.UNIT_METERS);
+
+    RouteStepProgress routeStepProgress
+      = new RouteStepProgress(firstLeg, 5, firstLeg.getSteps().get(4).getManeuver().asPosition());
+
+    Assert.assertEquals(stepDistance, routeStepProgress.getDistanceRemaining(), BaseTest.DELTA);
+  }
+
+  @Test
+  public void getDistanceRemaining_equalsCorrectValueAtIntervals() {
+    LineString lineString
+      = LineString.fromPolyline(firstStep.getGeometry(), Constants.PRECISION_6);
+    double stepDistance = TurfMeasurement.lineDistance(lineString, TurfConstants.UNIT_METERS);
+
+    double stepSegments = 5; // meters
+
+    // Chop the line in small pieces
+    for (double i = 0; i < stepDistance; i += stepSegments) {
+      Position position = TurfMeasurement.along(lineString, i, TurfConstants.UNIT_METERS).getCoordinates();
+
+      LineString slicedLine = TurfMisc.lineSlice(Point.fromCoordinates(position),
+        Point.fromCoordinates(route.getLegs().get(0).getSteps().get(1).getManeuver().asPosition()), lineString);
+
+      double distance = TurfMeasurement.lineDistance(slicedLine, TurfConstants.UNIT_METERS);
+
+      RouteStepProgress routeStepProgress = new RouteStepProgress(firstLeg, 0, position);
+      Assert.assertEquals(distance, routeStepProgress.getDistanceRemaining(), BaseTest.DELTA);
+    }
+  }
+
+  @Test
+  public void getDistanceRemaining_equalsZeroAtEndOfStep() {
+    RouteStepProgress routeStepProgress = new RouteStepProgress(firstLeg, 3,
+      firstLeg.getSteps().get(4).getManeuver().asPosition());
+
+    Assert.assertEquals(0, routeStepProgress.getDistanceRemaining(), BaseTest.DELTA);
+  }
+
+  @Test
+  public void getDistanceTraveled() {
+
```

Comment:
Removed the empty test, they were generated when creating the class (should be re-added later on)